src/adapter/sources.ts

```diff
@@ -433,7 +433,7 @@ export class SourceContainer {
         source: sourceUrl,
         line: uiLocation.lineNumber,
         column: uiLocation.columnNumber - 1, // source map columns are 0-indexed
-        bias: SourceMapConsumer.LEAST_UPPER_BOUND,
+        bias: SourceMapConsumer.GREATEST_LOWER_BOUND,
```

Comment:
v1 first does GREATEST_LOWER, and if it doesn't return a result (there can be no mapping in that direction) it searches with LEAST_UPPER.

Also this can cause the breakpoint to shift up a line, I have wanted to check for that case and prefer a position that eventually maps to the same source line.

Reply:
I have been meaning to go through test repos in old v1 issues and maybe that will help find issues like that.

Reply:
Hm, okay. So basically the algorithm would be something like:

```py
positions = map.get_all_position_for(location).sort()
for position in positions:
  # prefer returning the first position on the source line
  if map.get_original_position(position).line == location.line:
    return location

return positions[0] # default to greatest_lower_bound
```

Reply:
Hm, okay. So basically the algorithm would be something like:

```py
positions = map.get_all_position_for(location).sort()
for position in positions:
  # prefer returning the first position on the source line
  if map.get_original_position(position).line == location.line:
    return location

return positions[0] # default to greatest_lower_bound
```